scripts/src/java/org/oppia/android/scripts/label/AccessibilityLabelCheck.kt

```diff
@@ -0,0 +1,156 @@
+package org.oppia.android.scripts.label
+
+import org.oppia.android.scripts.proto.AccessibilityLabelExemptions
+import org.w3c.dom.Node
+import org.w3c.dom.NodeList
+import java.io.File
+import java.io.FileInputStream
+import java.util.stream.Collectors
+import java.util.stream.IntStream
+import javax.xml.parsers.DocumentBuilderFactory
+
+/**
+ * Script for ensuring that all the Activities in the codebase are defined with accessibility
+ * labels.
+ *
+ * Usage:
+ *   bazel run //scripts:accessibility_label_check -- <path_to_directory_root> <path_to_app_level
+ *   _manifest_file>
+ *
+ * Arguments:
+ * - path_to_directory_root: directory path to the root of the Oppia Android repository.
+ * - path_to_app_level_manifest_file: path to the manifest file of app layer.
+ *
+ * Example:
+ *   bazel run //scripts:accessibility_label_check -- $(pwd) app/src/main/AndroidManifest.xml
+ */
+fun main(vararg args: String) {
+  val repoPath = "${args[0]}/"
+
+  val accessibilityLabelExemptiontextProto = "scripts/assets/accessibility_label_exemptions"
+
+  val accessibilityLabelExemptionList = loadAccessibilityLabelExemptionsProto(
+    accessibilityLabelExemptiontextProto
+  ).getExemptedActivityList()
+
+  val manifesFilePath = args[1]
+
+  val fullPathToManifestFile = repoPath + manifesFilePath
+
+  val builderFactory = DocumentBuilderFactory.newInstance()
+
+  val docBuilder = builderFactory.newDocumentBuilder()
+
+  val doc = docBuilder.parse(File(fullPathToManifestFile))
+  doc.getDocumentElement().normalize()
+
+  val completeActivityList = convertNodeListToListOfNode(doc.getElementsByTagName("activity"))
+
+  val activityListWithoutLabel = completeActivityList.filter { node ->
+    checkIfActivityHasMissingLabel(node, accessibilityLabelExemptionList)
+  }
+
+  logFailures(activityListWithoutLabel, repoPath, accessibilityLabelExemptiontextProto)
+
+  if (activityListWithoutLabel.isNotEmpty()) {
+    throw Exception("ACCESSIBILITY LABEL CHECK FAILED")
+  } else {
+    println("ACCESSIBILITY LABEL CHECK PASSED")
+  }
+}
+
+/**
+ * Checks whether an activity element has a missing label.
+ *
+ * @param activityNode instance of Node
+ * @param accessibilityLabelExemptionList list of all the exemptions of the label check
+ * @return whether the label is present or not
+ */
+private fun checkIfActivityHasMissingLabel(
+  activityNode: Node,
+  accessibilityLabelExemptionList: List<String>
+): Boolean {
+  val attributesList = activityNode.getAttributes()
+  val activityPath = attributesList.getNamedItem("android:name").getNodeValue()
+  if (
+    activityPath
+      .removePrefix(".")
+      .replace(".", "/") in accessibilityLabelExemptionList
```

Comment:
I suggest extracting the 'removePrefix.replace()' transformation to a separate variable so that the if-check can fit on one line (for better readability/code clarity).

Reply:
This is outdated.